### PR TITLE
protocol: Add a new protocol that supports send/receive a signal

### PR DESCRIPTION
**Pull request by audoan99**
This protocol includes a request and an event:

- Request: Client sends a signal "0" or "1" to Server. If it is "1", the server will increment a counter variable and print it to the console. Otherwise, if it is "0", the server will just print it to the console without increment the counter.
- Event: Server would send a signal "1" if it is received the request from Client. Otherwise, it would send a signal "0". The client checks the received event and print it to the console. 

* weston-simple.c : use this protocol to send/receive signals.

### CODE REVIEW

File: ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,78 @@
+project (simple-weston-signal)
+
+find_package(PkgConfig)
+pkg_check_modules(WAYLAND_CLIENT wayland-client REQUIRED)
+pkg_check_modules(WAYLAND_CURSOR wayland-cursor REQUIRED)
+pkg_check_modules(LIBWESTON_PROTOCOLS libweston-6-protocols QUIET)
```
Comment (vnguyentrong) [inline ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt]: we are using weston-11 version

File: ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,78 @@
+project (simple-weston-signal)
+
+find_package(PkgConfig)
+pkg_check_modules(WAYLAND_CLIENT wayland-client REQUIRED)
+pkg_check_modules(WAYLAND_CURSOR wayland-cursor REQUIRED)
```
Comment (vnguyentrong) [inline ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt]: Do we need wayland-cursor?

File: ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+add_dependencies(${PROJECT_NAME} ${LIBS})
+
+add_definitions(${DLT_CFLAGS})
+
+if(${DLT_FOUND})
+    add_definitions(-DDLT)
+endif(${DLT_FOUND})
+
+if(${LIBWESTON_PROTOCOLS_FOUND})
+    target_link_libraries(${PROJECT_NAME} pthread)
```
Comment (vnguyentrong) [inline ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt]: Is pthread need?

File: ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+static void
+signal_format(void *data,
+	      struct weston_simple *weston_simple,
+	      uint32_t signal_response)
+{
+        if(signal_response == 1)
+        {
+                printf("Server received a signal from client!\n");
+        }
```
Comment (vnguyentrong) [inline ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c:24]: violate coding rule

File: ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+static void
+registry_handle_global_remove(void *data, struct wl_registry *registry,
+                              uint32_t name)
+{
+
+}
+
+static const struct wl_registry_listener registry_listener = {
```
Comment (vnguyentrong) [inline ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c:60]: not consistent with above function

File: ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt (hunk 1 above)
Comment (audoan99) [reply ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt]: Yes, I updated it.

File: ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt (hunk 2 above)
Comment (audoan99) [reply ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt]: Yes, it's not really necessary.

File: ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt (hunk 3 above)
Comment (audoan99) [reply ivi-layermanagement-examples/simple-weston-signal/CMakeLists.txt]: Yes, it's not really necessary.

File: ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c (hunk 4 above)
Comment (audoan99) [reply ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c:24]: Yes, I fixed it according to the rule.

File: ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c (hunk 5 above)
Comment (audoan99) [reply ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c:60]: Yes, I added a destroy function to the above function. Struct wl_registry_listener need the address of 2 function registry_handle_global and registry_handle_global_remove.

Comment (vnguyentrong) [reply ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c:60]: sorry, this comment is in the wrong place, with the struct you can leave it as current, but with function, see:

```
static void
registry_handle_global_remove()
```

not consistent with:

`int init_wayland_context()`

Comment (audoan99) [reply ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c:60]: > `int init_wayland_context()`
Yes, I updated it.

**Pull request merged**